### DISCUSSION TITLE
There is a way to Restart/Shutdown the server from AdonisJS code ?

### DISCUSSION THREAD

**Discussion by wemersonrv** [adonisjs/core#1823] [Help] [answered]
Hello guys 

THere is a way to run SO shell commands as privileged user ?

I need to restart or shutdown the server, stuffs like that. There is a way to do that ?

**Comment by Banashek** [comment, marked as answer]
Could you elaborate a bit more?

Do you mean that you want to shut down or restart the operating system from the web application? Perhaps you could explain what your goal is if so.

One way to do that (if it's your goal) would be to just use node's `child_process.exec` to call the shell's shutdown command (`shutdown now` or `shutdown -r now` for some linuxes).

**Reply by wemersonrv** [reply]
@Banashek It's exactly that. Need to `shutdown` the machine.

**Reply by Banashek** [reply]
Something like [this article](https://ourcodeworld.com/articles/read/411/how-to-shutdown-and-reboot-linux-with-node-js) should work. Just put that wherever in your adonis application that it makes sense.

**Reply by wemersonrv** [reply]
> Something like [this article](https://ourcodeworld.com/articles/read/411/how-to-shutdown-and-reboot-linux-with-node-js) should work. Just put that wherever in your adonis application that it makes sense.

Hey man, tx for the article. This is exactly what i need. 

But the `shutdown` command must run as root or run from a sudo group and still have to enter a username / password. The user where the adonis app is is even part of the sudo group, but there is no way to enter the password.

Do you know if there is a way to run without having to enter a username and password?
